index.d.ts

```diff
@@ -273,13 +279,15 @@ declare namespace chalk {
 	}
 }
 
+declare type NoThis<T> = {[K in keyof T]: T[K] extends T ? T : T[K]};
```

Comment:
Can you please elaborate on this?

As far as I understand it, this will take all properties from `T` that subclass `T` and replace them with `T`. What properties/use case does this target?

Reply:
That without this, TypeScript considers the following valid:
```ts
import chalk = require('chalk');

// Will throw at runtime because `chalk.reset.supportsColor` is `undefined`:
chalk.reset.supportsColor.level;
```

This is because without this, TypeScript resolves `chalk.reset` to `typeof chalk` rather than `Chalk`.

Reply:
Why do we keep returning `this` then? Wouldn't it be more straightforward to simply return the `Chalk` interface instead of `this` (given `Chalk` and `ChalkFunction` are merged together)?

Reply:
Well, unless we were to use proxies to have the style builder properties return the object from which they’re requested, then yes, it’s probably more straightforward to just return `Chalk`.

Also, the following doesn’t work for whatever reason:
```js
class CustomChalk extends chalk.Instance {
	constructor(options) {
		super(options);
	}
}

new CustomChalk(); // Is of type Chalk and not CustomChalk
```

Also, `instanceof` checks don’t work either.

Reply:
The subclassing issue is a valid one, is this something that makes sense at all for Chalk? Any use cases? @sindresorhus

Reply:
IIRC the subclassing issue was purely for the case of attaching your own colors to chalk.